core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -606,8 +613,10 @@ private void syncExistingInodeMetadata(LockedInodePath inodePath, boolean skipLo
               // Only set group if not empty
               builder.setGroup(group);
             }
+//            SetAttributeContext ctx = SetAttributeContext.mergeFrom(builder)
+//                .setUfsFingerprint(ufsFingerprint);
             SetAttributeContext ctx = SetAttributeContext.mergeFrom(builder)
-                .setUfsFingerprint(ufsFingerprint);
+                    .setUfsFingerprint(ufsFpParsed.serialize());
```

Comment:
Convert to String only when necessary. It's a shame we store the fingerprint as String in the inode. If we store the obj format by creating a proto for Fingerprint, we can get rid of a lot more parsing.